src/Extensions/src/Eventuous.AspNetCore.Web/Http/CommandHttpApiBase'TResult.cs

```diff
@@ -46,12 +49,40 @@ protected async Task<ActionResult<TResult>> Handle<TCommand>(TCommand command, C
         return AsActionResult<TAggregate>(result);
     }
 
-    static ActionResult<TResult> AsActionResult<T>(Result result) where T : Aggregate
-        => result is ErrorResult error
+    protected virtual ActionResult<TResult> AsActionResult<T>(Result result) where T : Aggregate {
```

Comment:
Why does it need to be virtual?

Reply:
We need the option to override it. 
We have the following use case:
The `CommandService` is doing validation with FluentValidation. We're rewriting the result of FluentValidation to a `ValidationProblemDetails`. Our `Kiota` client is picking it up and binds the result to the frontend. So the override looks like this:

```C#
protected override ActionResult<TResult> AsActionResult<T>(Result result)
{
    return result is ErrorResult error
        ? error.Exception switch
        {
            ValidationException => MapValidationExceptionAsValidationProblemDetails(error),
            _ => base.AsActionResult<T>(result)
        }
        : base.AsActionResult<T>(result);
}

private BadRequestObjectResult MapValidationExceptionAsValidationProblemDetails(ErrorResult error)
{
    if (error is null || error.Exception is null || error.Exception is not ValidationException)
    {
        throw new ArgumentNullException(nameof(error), "Exception in result is not of the type `ValidationException`. Unable to map validation result.");
    }

    ValidationException exception = (ValidationException)error.Exception;

    var problemDetails = new ValidationProblemDetails()
    {
        Status = StatusCodes.Status400BadRequest,
        Detail = "Please refer to the errors property for additional details."
    };

    var groupFailures = exception.Errors.GroupBy(v => v.PropertyName);
    foreach (var groupFailure in groupFailures)
    {
        problemDetails.Errors.Add(groupFailure.Key, groupFailure.Select(s => s.ErrorMessage).ToArray());
    }

    return new BadRequestObjectResult(problemDetails);
}
```